products/BellHybrid/services/desktop/endpoints/deviceInfo/DeviceInfoEndpoint.cpp

```diff
@@ -23,13 +24,98 @@ namespace sdesktop::endpoints
     {
         switch (context.getMethod()) {
         case http::Method::get:
-            getDeviceInfo(context);
+            handleGet(context);
             break;
         default:
             context.setResponseStatus(http::Code::BadRequest);
-            sender::putToSendQueue(context.createSimpleResponse());
             break;
         }
+        sender::putToSendQueue(context.createSimpleResponse());
+    }
+
+    auto DeviceInfoEndpoint::handleGet(Context &context) -> void
+    {
+        const auto &requestBody = context.getBody();
+
+        if (requestBody[json::fileList].is_number()) {
+
+            const auto diagFileType = static_cast<DiagnosticFileType>(requestBody[json::fileList].int_value());
+
+            if (!magic_enum::enum_contains<DiagnosticFileType>(diagFileType)) {
+                LOG_ERROR("Bad diagnostic type %d requested", static_cast<unsigned>(diagFileType));
+
+                context.setResponseStatus(http::Code::BadRequest);
```

Comment:
please return from functions instead of setting "globals" (we can think of context as a "global void"), this way we would have hope to someday change this code to something better.

Reply:
Done